### ISSUE TITLE
Is circle softmax equal to original softmax?

### ISSUE THREAD

**Issue by TCHeish** [JDAI-CV/fast-reid#225]
Hi,

In Base-stringerbaseline.yml:
MODEL.HEADS.CLS_LAYER: "CircleSoftmax"
MODEL.LOSSED.NAME: ("CrossEntrpyLoss", "TripletLoss")

my question is:
1. The difference between circle softmax and softmax?
2. The total loss = CrossEntrpyLoss(with CircleSoftmax) + TripletLoss, is there any reason why using this combination.(what kinds of experience on this you have done, can you share ? )
3. In CVPR 2018 Bag of tricks paper, the triplet loss is applied before bn layer, now seems applied after bn layer, any explainable experience can share?

Thanks a lot!

**Comment by L1aoXingyu** [comment]
1. Circle softmax is just like arcface or cosface, you can name it margin-based softmax. The difference between softmax is the margin.

2. We call circle softmax "ID/classification loss", triplet loss "metric loss", these two type loss combination can boost performance.

3. Just through experiments, we found features after the bn layer can get better results when using `SBS` configuration.
